package/InternalEAWGen.roc

```diff
@@ -0,0 +1,140 @@
+# Examples:
+# 0000..001F     ; N    # The property may be followed by comments
+# 0020           ; Na
+parseLine : Str -> Result EawRange _
+parseLine = \line ->
+    listToParsingState = \rest -> { val: [], rest }
+
+    startHexBytes =
+        line
+        |> Str.toUtf8
+        |> listToParsingState
+        |> Helpers.takeHexBytes
+
+    endHexBytes =
+        when startHexBytes.rest is
+            ['.', '.', .. as rest] -> rest |> listToParsingState |> Helpers.takeHexBytes
+            _ -> startHexBytes # This range has length 1. The start will be repeated as the end
+
+    endHexBytes.rest
+    |> List.splitFirst ';'
+    |> Result.map \{ after } -> after
+    |> Result.try \commentdEAP -> List.splitFirst commentdEAP '#'
+    |> Result.map \{ before } -> before
+    |> Result.try Str.fromUtf8
+    |> Result.map Str.trim
+    |> Result.try \eawp ->
+        startHexBytes.val
+        |> Str.fromUtf8
+        |> Result.map \start -> (eawp, "0x$(start)")
+    |> Result.try \(eawp, start) ->
+        endHexBytes.val
+        |> Str.fromUtf8
+        |> Result.map \end -> (eawp, start, "0x$(end)")
+
+expect parseLine "0020           ; Na # Zs         SPACE" == Ok ("Na", "0x0020", "0x0020")
+expect parseLine "0025..0027     ; Na # Po     [3] PERCENT SIGN..APOSTROPHE" == Ok ("Na", "0x0025", "0x0027")
+
+parsedLines : List (Result EawRange _)
+parsedLines = file |> Str.split "\n" |> List.map parseLine
+
+originalRanges =
+    parsedLines
+    |> List.keepOks \x -> x
+
+# We can drop the Neutral ranges because those are the default
+optimizedRanges =
+    List.dropIf originalRanges \(eawp, _, _) -> eawp == "N"
+
+# The input file contains many consecutive ranges with the same property value.
+# We will merge them into wider ranges to work with a smaller number of ranges.
+mergedRanges =
+    headRes = List.first optimizedRanges
+    tail = List.dropFirst optimizedRanges 1
+    res =
+        when headRes is
+            Err _ -> crash "Something went wrong while parsing the input."
+            Ok head -> List.walk tail { merged: [], merging: head } mergeOp
+
+    List.append res.merged res.merging
+
+mergeOp : { merged : List EawRange, merging : EawRange }, EawRange -> { merged : List EawRange, merging : EawRange }
+mergeOp = \{ merged, merging }, currentValue ->
+    if merging.0 == currentValue.0 && (Helpers.hexStrToU32 merging.2) + 1 == (Helpers.hexStrToU32 currentValue.1) then
+        {
+            merged,
+            merging: (merging.0, merging.1, currentValue.2),
+        }
+    else
+        {
+            merged: List.append merged merging,
+            merging: currentValue,
+        }
+
+expect mergeOp { merged: [], merging: ("Na", "0x0020", "0x0020") } ("Na", "0x0021", "0x0021") == { merged: [], merging: ("Na", "0x0020", "0x0021") }
+expect mergeOp { merged: [], merging: ("Na", "0x0020", "0x0020") } ("Na", "0x0022", "0x0022") == { merged: [("Na", "0x0020", "0x0020")], merging: ("Na", "0x0022", "0x0022") }
```

Comment:
Pulled this out into a separate helper and added some unit tests because I was finding it hard to follow.